legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/junit/LegendSDLCTestSuiteBuilder.java

```diff
@@ -126,21 +128,52 @@ public TestSuite buildSuite(String name, Collection<? extends Entity> entitiesFo
         }
         EntityToPureConverter converter = new EntityToPureConverter();
         entitiesForTesting.stream()
-                .map(e ->
+                .flatMap(e ->
                 {
-                    Optional<TestSuiteBuilder> optionalBuilder = converter.fromEntityIfPossible(e).map(element -> testSuiteBuilders.get(element.getClass()));
-                    if (!optionalBuilder.isPresent())
+                    Optional<? extends PackageableElement> packageableElement = converter.fromEntityIfPossible(e);
+                    Optional<TestSuiteBuilder> genericTestBuilder = packageableElement.map(element -> testSuiteBuilders.get(PackageableElement.class));
+                    Optional<TestSuiteBuilder> specificTestBuilder = packageableElement.map(element -> testSuiteBuilders.get(element.getClass()));
+
+                    if (!(genericTestBuilder.isPresent() && TestableRunnerExtensionLoader.isRunnerProvidedForClassifierPath(e.getClassifierPath())) && !specificTestBuilder.isPresent())
```

Comment:
This seems pretty inefficient, especially since it's going to be called again below. Surely it would be better to get the list of extensions once and index them by what they support?

Reply:
Done